sql/fxa_users_last_seen_v1.sql

```diff
@@ -0,0 +1,47 @@
+WITH
+  current_sample AS (
+  SELECT
+    -- Include dummy dates for date_last_* fields to make schema match with `previous`.
+    DATE '2000-01-01' AS date_last_seen,
+    DATE '2000-01-01' AS date_last_seen_in_tier1_country,
+    * EXCEPT (submission_date,
+      generated_time)
+  FROM
+    fxa_users_daily_v1
+  WHERE
+    submission_date = @submission_date ),
+  previous AS (
+  SELECT
+    * EXCEPT (submission_date,
+      generated_time)
+  FROM
+    fxa_users_last_seen_v1
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 1 DAY)
+    AND date_last_seen > DATE_SUB(@submission_date, INTERVAL 28 DAY) )
+SELECT
+  @submission_date AS submission_date,
+  CURRENT_DATETIME() AS generated_time,
+  -- Record the last day on which we recieved any core ping at all from this user.
+  IF(current_sample.user_id IS NOT NULL,
+    @submission_date,
+    previous.date_last_seen) AS date_last_seen,
```

Comment:
I feel like we should use `COALESCE(current_sample.date_last_seen, previous.date_last_seen) AS date_last_seen` and do `@submission_date AS date_last_seen` on line 5

Reply:
I'm going to think some more about this. It might make sense to wrap up some logic in a SQL UDF to make sure we're consistent between the different date_last_seen columns.

Reply:
As I played with this, I better understood how your COALESCE proposal fits together and I've gone with that rather than a UDF.